elements/lisk-tree/src/merkle_tree/merkle_tree.ts

```diff
@@ -267,6 +282,176 @@ export class MerkleTree {
 		return rightWitness;
 	}
 
+	public async getSiblingHashes(idxs: ReadonlyArray<number>): Promise<Buffer[]> {
+		if (this._size === 0) {
+			return [];
+		}
+
+		const treeHeight = Math.ceil(Math.log2(this._size)) + 1;
+
+		const sortedIndexes: NodeIndex[] = [];
+		for (const index of idxs) {
+			const serializedIndexBinaryString = index.toString(2);
+			const indexBinaryString = serializedIndexBinaryString.substring(
+				1,
+				serializedIndexBinaryString.length,
+			);
+			const location = {
+				nodeIndex: parseInt(indexBinaryString, 2),
+				layerIndex: treeHeight - indexBinaryString.length,
+			};
+
+			sortedIndexes.push(location);
+		}
+		// Remove empty indexes
```

Comment:
I would do this before nin the loop of line 293. Just check there directly the encoded index against a constant flag (00 hex), and in case it is equal, just continue